### ISSUE TITLE
CGD1 QingPing Alarm Clock with sensors failed to detect

### ISSUE THREAD

**Issue by matthiufungho** [custom-components/sensor.mitemp_bt#80]
Hello,

I have the QingPing alarm clock (CGD1) but having problem receiving sensor message on HA.  component debug log shows the 0 xiaomi device found.  The device looks like this:
https://www.gearbest.com/blog/new-gear/xiaomi-qingping-bluetooth-alarm-clock-review-9838

I am not using the MiHome to get the encryptor key.  The QingPing alarm clock works with the qingping+ app.  i use the Stream app to sniff the http requests and found the following json:

```
:57
{
        "pair_type" : 0,
        "sn" : "",
        "mac" : "582xxxxxxx696",
        "pair_id" : "5edbxxxxxxxxxxxxxx6d9d7",
        "version" : "1.0.1_0067",
        "link_token" : "c91ee3xxxxxxxxxxxxxxxxx3933e7",
        "_id" : "5edbxxxxxxxxxxxxa16d9d7",
        "rank" : 2,
        "boot_version" : "",
        "widget_rank" : 4,
        "device_id" : "DADAAxxxxxxxxxxxxxxxxxxxx58",
        "model" : "Dove",
        "name" : "Bedroom Alarm  Clock",
        "widget_status" : 0
      }
```
Here is the configuration.yaml
```
sensor:
  - platform: mitemp_bt
    rounding: True
    decimals: 1
    period: 60
    log_spikes: False
    use_median: False
    active_scan: False
    hci_interface: 0
    batt_entities: True
    encryptors:
#        '58:2D:34:51:D6:96': 'C91EExxxxxxxxxxxxxxxxxxxxxxxxx33E7'
        '58:2D:34:51:D6:96': 'DADAAxxxxxxxxxxxxxxxxxxxxxxxxD58'
    report_unknown: False
    whitelist: False
```

```
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] Time to analyze...
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] Getting data from HCIdump thread
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] HCIdump thread: joining
2020-06-06 16:39:08 DEBUG (Thread-16) [custom_components.mitemp_bt.sensor] HCIdump thread: main event_loop stopped, finishing
2020-06-06 16:39:08 DEBUG (Thread-16) [custom_components.mitemp_bt.sensor] HCIdump thread: Run finished
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] HCIdump thread: joined
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] Spawning HCIdump thread(s).
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] HCIdump thread: Init
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] HCIdump thread: Init finished
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] Starting HCIdump thread for hci0
2020-06-06 16:39:08 DEBUG (Thread-17) [custom_components.mitemp_bt.sensor] HCIdump thread: Run
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] HCIdump threads count = 1
2020-06-06 16:39:08 DEBUG (Thread-17) [custom_components.mitemp_bt.sensor] HCIdump thread: Connection
2020-06-06 16:39:08 DEBUG (Thread-17) [custom_components.mitemp_bt.sensor] HCIdump thread: Connected
2020-06-06 16:39:08 DEBUG (Thread-17) [custom_components.mitemp_bt.sensor] HCIdump thread: start main event_loop
2020-06-06 16:39:08 DEBUG (SyncWorker_11) [custom_components.mitemp_bt.sensor] Finished. Parsed: 785 hci events, 0 xiaomi devices.
```
Any help would be greatly appreciated!!

**Comment by matthiufungho** [comment]
Mihome does not work on iPhone but I am able to find Android to have MiHome installed and added the QingPing Alarm Clock.  However, I am not able to find the bind_key using Packet Capture on Android.

**Comment by Magalex2x14** [comment]
Hello! This device sends out messages in two formats:
1. Unencrypted messages for working with the QingPing application. These messages are not (yet) supported by this custom component.
2. Encrypted messages for Mija ecosystem (MiHome app).

That is, to receive messages from this device, you need to carry out the pairing procedure with the MiHome application, and only the key intercepted during this process is suitable.
Too bad you failed with PacketCapture. You can try an easier way on android using a [modified application](https://github.com/custom-components/sensor.mitemp_bt/issues/7#issuecomment-595874419).

Why doesn’t MiHome application work on your iPhone? I have never had a problem with it on my iPhone.

**Comment by matthiufungho** [comment]
Thanks @Magalex2x14!

I tried MiHome on iPhone again.  The QingPing Clock was not in the device list.  I changed Profile->Settings->Region from Hong Kong to Mainland China.  Search the device list again and I found the QingPing Clock.  Stream sniff during bluetooth pairing with VPN switched on and I was able to find bltind http POST.  Decode the request JSON and got the bind_key.  Use the value in configuration.yaml and now I am able to see mi_b_MAC, mi_t_MAC and mi_h_MAC sensors.

It works great! 

Sorry for the trouble.
